### PR TITLE
Added support for blob rehydrate priority on blob properties

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy [hunk 1]
```diff
@@ -1317,6 +1345,32 @@ class ContainerAPITest extends APISpec {
         cc.listBlobs(new ListBlobsOptions().setMaxResultsPerPage(PAGE_SIZE), null).stream().count() == NUM_BLOBS
     }
 
+    @Unroll
+    def "List blobs hier rehydrate priority"() {
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy:1349]: typo hier

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy:1349]: Thats just short for hierarchical

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy:1349]: hahahahaha gotcha

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy [hunk 2]
```diff
@@ -949,6 +949,29 @@ class BlobAPITest extends APISpec {
         bc.getPropertiesWithResponse(null, null, null).getStatusCode() == 200
     }
 
+    @Unroll
+    def "Get properties rehydrate priority"() {
+        setup:
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy:954]: Putting this test here feels a little disjointed to me. It seems like we should add this property to the list in the test above and then also add a getProperties call to the tests for setTier that test rehydratePriority (which I'm not actually finding any that test that?)

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy:954]: Ok, I can make this a setTier test and just check that the rehydrate priority is null on a regular get properties call

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy:955]: Modified the test

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy [hunk 3]
```diff
@@ -942,13 +942,37 @@ class BlobAPITest extends APISpec {
...
+        properties.getRehydratePriority() == null // tested in setTier rehydrate priority
     }
 
     def "Get properties min"() {
         expect:
         bc.getPropertiesWithResponse(null, null, null).getStatusCode() == 200
     }
 
+    @Unroll
+    def "Get properties rehydrate priority"() {
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobAPITest.groovy:954]: whoops forgot to delete this

**Pull request merged**